### PR TITLE
fix(discover): Fix broken query when clicking tags in eventDetails

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This fixes an issue where a user adds a custom tag of the same name as a key from contexts (eg. device.uuid). This currently works because most of these tags come out of normalizing the event, which pulls the context into tags, so the values for both are the same.

### Screenshot
Screen Shot 2020-06-26 at 10 03 40 AM
Screen Shot 2020-06-26 at 10 04 17 AM

### CODE REVIEW

**Review by evanh — APPROVED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx [hunk 1]
```diff
@@ -106,6 +106,13 @@ class EventDetailsContent extends AsyncComponent<Props, State> {
     return this.props.eventSlug.split(':')[0];
   }
 
+  generateTagKey = (tag: EventTag) => {
+    // Some tags may be normalized from context, but not all of them are.
+    // This supports a user making a custom tag with the same name as one
+    // that comes from context as all of these are also tags.
+    return `tags[${tag.key}]`;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx:113]: Could we quote only when the tag overlaps a known field? The field list in `utils/discover/fields` could be used.

Comment (k-fish) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx:113]: like only quote on those? and not any other custom tag? that's a good idea 👍

Comment (markstory) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx:113]: Yes. The quote syntax is only required when tags overlap with core event properties. Doing it all the time clutters the search bar.

Comment (k-fish) [reply src/sentry/static/sentry/app/views/eventsV2/eventDetails/content.tsx]: Fixed

**Review by markstory — APPROVED** [review]

**Pull request merged**